### PR TITLE
Fire alarm sounds

### PR DESCRIPTION
**Pull request by kugamo**
## What Does This PR Do

Adds a sound for when fire alarms are pulled. All the alarms play in sync, so there won't be a cacophony of alarms going off all at slightly different times... drowning everything else out. Also I added them to their own sound channel and to the volume mixer, so players can choose a less loud or complete silence if they want.

## Why It's Good For The Game

Fire alarms without an alarm!? Its great flavour for the setting, adds atmosphere and realism to the game.

## Images of changes

https://user-images.githubusercontent.com/16618648/174395831-fb935e89-7827-42f9-9e2c-4000bcc40ac1.mp4

## Changelog
:cl:
soundadd: Added sounds for fire alarms.
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
I am torn on this one.


- On the one hand, yes, immersion
- On the other hand, I just see this as a way to piss people off, as well as the sheer noise when a plasmaflood is going down

**Comment by TrashDoxx** [comment]
I need option in sound mixer.

**Comment by kugamo** [comment]
> I am torn on this one.
> 
>     * On the one hand, yes, immersion
> 
>     * On the other hand, I just see this as a way to piss people off, as well as the sheer noise when a plasmaflood is going down

> I need option in volume mixer.

I've made changes. Now all the alarms sound in sync, at least eliminating the potential for it to be a cacophony of alarms. Also added it as an option to the volume mixer, so players who are easily disturbed by it can just turn it off.. and those with sensitive ears to tune it to their liking.

**Comment by Bm0n** [comment]
It might be best to use goons firealarm sound instead, TG used it before they replaced the sound with an arguably even worse and looping one. The goon alarm plays once on activation and then doesn't loop since looping is very annoying imo.

https://user-images.githubusercontent.com/92271472/174401617-7022d5f3-35cc-495c-a58b-705a02d59004.mp4

**Comment by lewcc** [comment]
> It might be best to use goons firealarm sound instead, TG used it before they replaced the sound with an arguably even worse and looping one. The goon alarm plays once on activation and then doesn't loop since looping is very annoying imo. The OGG can be found in older TG code or goon code
> 
>  goon_sound_machinery_FireAlarm.mp4

Eh, the goon one is particularly very rough on the ears, especially in the case where there's a plasmafire or bomb and everything is just high-pitched beeping.

File: code/datums/looping_sounds/machinery_sounds.dm [hunk 1]
```diff
@@ -17,3 +17,19 @@
...
+	mid_length = 20
+	volume = 80
+	extra_range = 15
+	falloff_exponent = 5
+	channel = CHANNEL_FIREALARM
+
+/datum/looping_sound/firealarm/sound_loop(looped)
+	. = ..()
+	if(!length(output_atoms))
+		src.stop()
```
Comment (SteelSlayer) [inline code/datums/looping_sounds/machinery_sounds.dm:34]:
```suggestion
		stop()
```

**Review by SteelSlayer — APPROVED** [review]

**Comment by AffectedArc07** [comment]
> It might be best to use goons firealarm sound instead, TG used it before they replaced the sound with an arguably even worse and looping one. The goon alarm plays once on activation and then doesn't loop since looping is very annoying imo. The OGG can be found in older TG code or goon code
> 
>  goon_sound_machinery_FireAlarm.mp4

Yeah not this. This sounds too much like my real fire alarm, and PTSD triggers whenever I hear that.

**Pull request merged**